src/Common/Common.Api/Gateways/Interfaces/IGatewayClient.cs

```diff
@@ -25,5 +25,15 @@ public interface IGatewayClient
         Task<HttpResponseMessage> UpdateTenantPolicy(UpdateGatewayPolicyRequest request);
         Task<IEnumerable<InstallerPrincipal>> GetInstallerPrincipals(GatewayType? type);
         Task<HttpResponseMessage> UpdateInstallerPrincipals(UpdateGatewayInstallersRequest request);
+        Task<IEnumerable<GatewayClusterDatasource>> GetGatewayClusterDatasources(Guid gatewayClusterId, bool asIndividual);
+        Task<GatewayClusterDatasource> GetGatewayClusterDatasource(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, bool asIndividual);
+        Task<HttpResponseMessage> GetGatewayClusterDatasourceStatus(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, bool asIndividual);
+        Task<GatewayClusterDatasource> CreateGatewayClusterDatasource(Guid gatewayClusterId, PublishDatasourceToGatewayClusterRequest datasourceToGatewayClusterRequest, bool asIndividual);
+        Task<HttpResponseMessage> UpdateGatewayClusterDatasource(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, UpdateGatewayClusterDatasourceRequest updateDatasourceRequest, bool asIndividual);
+        Task<HttpResponseMessage> DeleteGatewayClusterDatasource(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, bool asIndividual);
+        Task<HttpResponseMessage> AddUsersToGatewayClusterDatasource(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, UserAccessRightEntry datasourceUser, bool asIndividual);
+        Task<IEnumerable<UserAccessRightEntry>> GetGatewayClusterDatasourceUsers(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, bool asIndividual);
+        Task<HttpResponseMessage> RemoveGatewayClusterDatasourceUser(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, string user, bool asIndividual);
+        Task<IEnumerable<DatasourceErrorDetails>> UpdateGatewayClusterDatasourceCredentials(Guid gatewayClusterId, Guid gatewayClusterDatasourceId, DatasourceCredentialDetails updateCredentialDetails, bool asIndividual);
```

Comment:
UpdateGatewayClusterDatasourceCredentialRequest

Reply:
I would like to keep this the same as in the backend code.